nautobot/core/tests/test_api.py

```diff
@@ -780,23 +783,32 @@ def setUpTestData(cls):
             "DateTimeField": "created",
         }
 
+    def _validate_sorted_response(self, response, queryset, field_name, is_fk_field=False):
+        self.assertHttpStatus(response, 200)
+
+        # If the field is a foreign key field, we cannot guarantee the relative order of objects with the same value across multiple objects.
+        # Therefore, we directly compare the names of the foreign key objects.
+        if is_fk_field:
+            api_data = list(map(lambda p: p[field_name]["name"] if p[field_name] else None, response.data["results"]))
+            queryset_data = list(queryset.values_list(f"{field_name}__name", flat=True)[:10])
```

Comment:
I don't really understand how this part works or the need for it. If the FK has the same value (`id`), surely it will also have the same `name`? It seems like this branch is asserting that the *nested* objects are ordered consistently, instead of asserting that the base objects are ordered consistently - is that the intent, or what am I misunderstanding?